src/main/groovy/edu/oregonstate/mist/cardsapi/resources/CardsResource.groovy

```diff
@@ -65,4 +82,129 @@ class CardsResource extends Resource {
+            List<String> invalidRarities = rarities - validRarities
+            if(invalidRarities) {
+                return badRequest("Invalid rarities: \'${invalidRarities.join("\', \'")}\'. " +
+                        "Valid rarities are: " +
+                        "basic, common, uncommon, rare").build()
+            }
+        }
+
+        if(!(name.or("")).matches(regEx)) {
+            return badRequest("Invalid name: \'" + name.get() +
+                    "\'. Name must match pattern: " +
+                    regEx).build()
+        }
+
+        for(int i = 0; i < keywords.size(); i++) {
+            if(!keywords[i].matches(regEx)) {
+                return badRequest("Invalid keyword: \'" + keywords[i] +
+                        "\'. All keywords " +
+                        "must match pattern: " + regEx).build()
+            }
+        }
+
+        List<Card> cards = cardFluent.getCards(types, name.orNull(), colors, rarities,
+                energyMin.or(0), energyMax.or(999),
+                keywords, number.or(10), isRandom.or(true))
+
+        ResultObject cardResult = cardsResult(cards)
+        ok(cardResult).build()
+    }
+
+    @POST
+    @Path('')
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    Response postCard (@Valid Card newCard) {
+
+        Response response = cardValidator(newCard)
+        if(response) {
+            return response
+        }
+
+        Integer id = cardDAO.getNextId()
+        cardDAO.postCard(
+                id,
+                newCard.type,
+                newCard.name,
+                newCard.color,
+                newCard.rarity,
+                newCard.energy,
+                newCard.description)
+
+        Card card = cardDAO.getCardById(id)
+        ResultObject cardResult = cardsResult(card)
+        created(cardResult).build()
+    }
+
+    Response cardValidator(Card card) {
+        if(!validTypes.contains(card.type)) {
+            return badRequest("Invalid type. " +
+                    "Valid types are skill, attack, power, status, curse").build()
+        }
+        if(!validColors.contains(card.color)) {
+            return badRequest("Invalid color. " +
+                    "Valid colors are red, green, blue, colorless").build()
+        }
+        if(!validRarities.contains(card.rarity)) {
+            return badRequest("Invalid rarity. " +
+                    "Valid rarities are basic, common, uncommon, rare").build()
+        }
+        if(!card.name.matches(regEx)) {
+            return badRequest("Invalid name: \'" + card.name +
+                    "\'. Name must match pattern: " +
+                    regEx).build()
+        }
+        if(!card.description.matches(regEx)) {
+            return badRequest("Invalid description: \'" + card.description +
+                    "\'. Description must match pattern: " +
+                    regEx).build()
+        }
+        if(!(card.energy >= 0 && card.energy <= 999)) {
```

Comment:
perhaps these numbers should be set as variables for the class? Then you could reference them in your error message too like `"Energy must be between ${minimumEnergy} and ${maximumEnergy}"` for example